exercises/webapp/src/components/HackerNews.vue

```diff
@@ -15,39 +20,50 @@ import NewsForm from './NewsForm.vue';
 export default {
   name: 'HackerNews',
   components: {NewsItem, NewsForm},
+  props: {
+    initialNews: {type: Array, required: true}
+  },
   data() {
     return {
-      items: [{
-        id: 0,
-        title: "Start Message",
-        voteCount: 0,
-      }],
-      id: 1
-    }
+      ascending: false,
+      items: [...this.initialNews],
+    };
   },
   computed: {
     sortedItems() {
-      return [...this.items].sort((o1, o2) => o2.voteCount - o1.voteCount)
+      let sortedArray;
+      if (this.ascending) {
+        sortedArray = [...this.items].sort((o2, o1) => o2.voteCount - o1.voteCount);
+      } else {
+        sortedArray = [...this.items].sort((o1, o2) => o2.voteCount - o1.voteCount);
+      }
+      return sortedArray;
+    },
+    nextId() {
+      return Math.max(...this.items.map(item => item.id), 0) + 1;
     }
   },
   methods: {
     onNewsAdd(newsTitle) {
       let newsItem = {
-        id: this.id++,
+        id: this.nextId,
         title: newsTitle,
         voteCount: 0
       };
       this.items.push(newsItem);
     },
     onNewsRemove(newsItem) {
-      this.items = this.items.filter((element) => element.id !== newsItem.id)
+      this.items = this.items.filter((element) => element.id !== newsItem.id);
     },
     onVoteChange(args) {
-      let item = this.items.find((element) => element.id === args.newsItem.id)
+      let item = this.items.find((element) => element.id === args.newsItem.id);
```

Comment:
You can also make it with .map
like this: this.news = this.news.map((x) => (x.id == newItem.id ? newItem : x));